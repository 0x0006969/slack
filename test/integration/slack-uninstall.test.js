const request = require('supertest');

const helper = require('.');
const fixtures = require('../fixtures');

describe('Uninstalling the slack app', () => {
  beforeEach(async () => {
    const { Installation, SlackWorkspace, Subscription } = helper.robot.models;
    // Create an installation
    const installation = await Installation.create({
      githubId: 1,
      ownerId: fixtures.org.id,
    });

    const slackWorkspace = await SlackWorkspace.create({
      slackId: 'T0001',
      accessToken: 'xoxp-token',
    });

    await Subscription.create({
      githubId: 1234,
      channelId: 'C0012',
      slackWorkspaceId: slackWorkspace.id,
      installationId: installation.id,
    });

    await Subscription.create({
      githubId: 4321,
      channelId: 'C0234',
      slackWorkspaceId: slackWorkspace.id,
      installationId: installation.id,
    });

    await Subscription.create({
      githubId: 5678,
      channelId: 'C0012',
      slackWorkspaceId: slackWorkspace.id,
      installationId: installation.id,
    });
  });
  test('deletes all corresponding subscriptions', async () => {
    const { Subscription } = helper.robot.models;
    expect(await Subscription.count()).toBe(3);

    await request(helper.probot.server).post('/slack/events')
      .send({
        token: process.env.SLACK_VERIFICATION_TOKEN,
        team_id: 'T0001',
        event: {
          type: 'app_uninstalled',
        },
        type: 'event_callback',
      })
      .expect(200)

    expect(await Subscription.count()).toBe(0);
  });
});
