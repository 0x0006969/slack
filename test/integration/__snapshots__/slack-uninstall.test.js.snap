// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uninstalling the slack app deletes all corresponding subscriptions 1`] = `
Object {
  "channelId": "C0012",
  "createdAt": 2019-03-26T10:44:06.646Z,
  "creatorId": "1",
  "deletedAt": 2019-03-26T10:44:06.709Z,
  "githubId": "1234",
  "githubName": "foo/repo1",
  "id": "1",
  "installationId": "1",
  "reason": "slack app uninstalled",
  "settings": Object {
    "commits": true,
    "pulls": true,
  },
  "slackWorkspaceId": "1",
  "type": "repo",
  "updatedAt": 2019-03-26T10:44:06.719Z,
}
`;

exports[`Uninstalling the slack app deletes all corresponding subscriptions 2`] = `
Object {
  "channelId": "C0234",
  "createdAt": 2019-03-26T10:44:06.653Z,
  "creatorId": "1",
  "deletedAt": 2019-03-26T10:44:06.712Z,
  "githubId": "4321",
  "githubName": "foo/repo2",
  "id": "2",
  "installationId": "1",
  "reason": "slack app uninstalled",
  "settings": Object {
    "commits": true,
    "pulls": true,
  },
  "slackWorkspaceId": "1",
  "type": "repo",
  "updatedAt": 2019-03-26T10:44:06.722Z,
}
`;

exports[`Uninstalling the slack app deletes all corresponding subscriptions 3`] = `
Object {
  "channelId": "C0012",
  "createdAt": 2019-03-26T10:44:06.656Z,
  "creatorId": "1",
  "deletedAt": 2019-03-26T10:44:06.714Z,
  "githubId": "5678",
  "githubName": "foo/repo3",
  "id": "3",
  "installationId": "1",
  "reason": "slack app uninstalled",
  "settings": Object {
    "commits": true,
    "pulls": true,
  },
  "slackWorkspaceId": "1",
  "type": "repo",
  "updatedAt": 2019-03-26T10:44:06.724Z,
}
`;

exports[`Uninstalling the slack app deletes all corresponding subscriptions 4`] = `Set {}`;

exports[`Uninstalling the slack app deletes all corresponding subscriptions 5`] = `Map {}`;
